Issue running demos on Windows (path/rootDir issue)

### Describe the bug

While trying to run `npm start` **on Windows** in the demo folders, which runs `node ../../scripts/run-demo-server.js` it throws an error in the following line:
https://github.com/Esri/arcgis-rest-js/blob/main/scripts/get-package-json.js#L19

The problem is that the build path in `rootDir` has a semicolon (something like `/D:/..`), and it is causing an exception in Linux based shells like Gitbash.

### Reproduction

You will probably need a Windows machine a follow any of the instructions (e.g. [job demo instructions](https://github.com/Esri/arcgis-rest-js/tree/0e410dc16e0dd2961affb09ff7efbfb9b6c4999a/demos/job))

### Logs

```shell
Sorry, I don't have them.
```


### System Info

```shell
Windows 11.
It was this repo at this point, so ArcGIS REST JS Version: 4.1.4
```


### Additional Information

Sorry, I can not provide more context. It was an easy fix, we just removed the semicolon locally, but I don't have time to submit a PR right now, but I wanted to at least issue this.

I do get an error when run:

```
cd demos/geocoder-browser
npm i
npm start
```

... on Windows 10 in Powershell.

The error I get:
```
> @esri/arcgis-rest-geocoder-vanilla@3.3.0 start
> node ../../scripts/run-demo-server.js

node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received null
    at new NodeError (node:internal/errors:405:5)
    at validateString (node:internal/validators:162:11)
    at join (node:path:429:7)
    at getPackages (file:///C:/_work/arcgis-rest-js/scripts/get-package-json.js:19:37)
    at async file:///C:/_work/arcgis-rest-js/scripts/run-demo-server.js:7:20 {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v18.17.1
npm ERR! Lifecycle script `start` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: @esri/arcgis-rest-geocoder-vanilla@3.3.0
npm ERR!   at location: C:\_work\arcgis-rest-js\demos\geocoder-browser
```

I agree with @hhkaos that we need to get the script at https://github.com/Esri/arcgis-rest-js/blob/main/scripts/get-package-json.js#L19 to be more Windows friendly.

--- a/demos/geocoder-browser/README.md
+++ b/demos/geocoder-browser/README.md
@@ -1,12 +1,12 @@
 # Running this demo
 
-1. Make sure you run `npm run bootstrap` in the root folder to setup the dependencies
+1. Make sure you run `npm run build` in the root folder to setup the dependencies
 2. Register a new app on https://developers.arcgis.com
-3. Add a redirect URL of `http://localhost:8080/post-sign-in.html` to your app.
+3. Add a redirect URL of `https://localhost:8080/authenticate.html` to your app.
 4. Copy the `config.js.template` file, rename it to `config.js`
 5. Copy your apps client id into your new `config.js` file.
 6. `npm start`
-7. Visit http://localhost:8080 and click "Sign In" to start the OAuth 2.0 process.
+7. Visit https://localhost:8080 and click "Sign In" to start the OAuth 2.0 process.
 8. Start clicking buttons to send off authenticated geocoding requests.
 
 **Note:** The server starts with a special configuration to serve URLs starting with `@esri/arcgis-rest-*` from their respective packages. In your application you will need to change these URLs to point to their respective locations.

--- a/demos/geocoder-browser/index.html
+++ b/demos/geocoder-browser/index.html
@@ -55,7 +55,7 @@ var authBtn = document.getElementById('sign-in');
 authBtn.addEventListener('click', function (e) {
   arcgisRest.ArcGISIdentityManager.beginOAuth2({
     clientId: ClientId,
-    redirectUri: "http://localhost:8080/authenticate.html",
+    redirectUri: "https://localhost:8080/authenticate.html",
     popup: true,
     portal: "https://www.arcgis.com/sharing/rest"
   }).then((session) => {

--- a/scripts/get-package-json.js
+++ b/scripts/get-package-json.js
@@ -1,9 +1,10 @@
 import { readFile } from "fs/promises";
 import { globby } from "globby";
 import pkgDir from "pkg-dir";
-import { join, dirname } from "path";
+import { posix } from "path";
+import { fileURLToPath } from "url";
 
-const __dirname = dirname(new URL(import.meta.url).pathname);
+const __dirname = fileURLToPath(new URL('.', import.meta.url)).replace(/\\/g, '/');
 
 /**
  * Returns an object like:
@@ -15,8 +16,10 @@ const __dirname = dirname(new URL(import.meta.url).pathname);
  * For all packages in the packages/* folder.
  */
 export default async function getPackages() {
-  const rootDir = await pkgDir(__dirname);
-  const packageFiles = await globby(join(rootDir, "packages/*/package.json"));
+  const rootDir = (await pkgDir(__dirname)).replace(/\\/g, "/");
+
+  const packageFiles = await globby(posix.join(rootDir, "packages/*/package.json"));
+
   return Promise.all(
     packageFiles.map((pkgPath) => {
       return readFile(pkgPath).then((pkg) => {

--- a/scripts/run-demo-server.js
+++ b/scripts/run-demo-server.js
@@ -1,15 +1,16 @@
 import browserSync from "browser-sync";
-import { join, resolve, dirname } from "path";
+import { resolve, posix } from "path";
+import { fileURLToPath } from "url";
 import getPackages from "./get-package-json.js";
 
 (async () => {
   const cwd = process.cwd();
   const packages = await getPackages();
-  const __dirname = dirname(new URL(import.meta.url).pathname);
+  const __dirname = fileURLToPath(new URL('.', import.meta.url)).replace(/\\/g, '/');
   const packageFolders = packages.map(({ name }) => {
     return {
       route: `/${name}`,
-      dir: resolve(join(__dirname, "../", "node_modules", name))
+      dir: resolve(posix.join(__dirname, "../", "node_modules", name))
     };
   });
 

